GliaWidgets/SecureConversations/SecureConversations.SendMessageButton.swift

```diff
@@ -0,0 +1,158 @@
+                title: "Title",
+                titleFont: .systemFont(ofSize: 15),
+                forgroundColor: .white,
+                backgroundColor: .black,
+                borderColor: .red,
+                borderWidth: 3,
+                cornerRadius: 8,
+                acitivityIndicatorColor: .green,
+                isActivityIndicatorShown: false
+            )
+        ) {
+            didSet {
+                renderProps()
+            }
+        }
+
+        lazy var messageTitleLabel = UILabel().makeView()
+        lazy var activityIndicator: ActivityIndicatorView = {
+            let indicator = ActivityIndicatorView()
+            indicator.isHidden = true
+            return indicator
+        }()
+        lazy var messageTitleStackView = UIStackView(arrangedSubviews: [
+            activityIndicator,
+            messageTitleLabel
+        ]).makeView { stackView in
+            stackView.axis = .horizontal
+            stackView.alignment = .center
+            stackView.spacing = 11
+        }
+
+        override init(frame: CGRect) {
+            super.init(frame: frame)
+            setup()
+            renderProps()
+        }
+
+        @available(*, unavailable)
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        func setup() {
+            addSubview(messageTitleStackView)
+            NSLayoutConstraint.activate([
+                messageTitleStackView.centerXAnchor.constraint(equalTo: centerXAnchor),
+                messageTitleStackView.centerYAnchor.constraint(equalTo: centerYAnchor)
+            ])
+        }
+
+        func renderProps() {
+            let props = UIKitProps(props)
+            messageTitleLabel.text = props.title
+            messageTitleLabel.font = props.titleFont
+            messageTitleLabel.textColor = props.forgroundColor
+            backgroundColor = props.backgroundColor
+            layer.borderWidth = props.borderWidth
+            layer.borderColor = props.borderColor.cgColor
+            layer.cornerRadius = props.cornerRadius
+            activityIndicator.color = props.acitivityIndicatorColor
+            renderedIsIndicatorShown = props.isActivityIndicatorShown
+        }
+
+        var renderedIsIndicatorShown: Bool = false {
+            didSet {
+                guard renderedIsIndicatorShown != oldValue else { return }
+                activityIndicator.isHidden = !renderedIsIndicatorShown
+
+                if renderedIsIndicatorShown {
+                    activityIndicator.startAnimating()
+                } else {
+                    activityIndicator.stopAnimating()
+                }
+            }
+        }
+    }
+}
+
+extension SecureConversations.SendMessageButton {
+    enum Props: Equatable {
+        case normal(NormalState)
+        case disabled(DisabledState)
+    }
+}
+
+extension SecureConversations.SendMessageButton.Props {
+    struct NormalState: Equatable {
```

Comment:
What is the difference between `NormalState` struct and `DisabledState` one. Looks like they have identical properties

Reply:
Will it be convenient to leave one struct `State` and still have different properties in Props?
```swift
let nornalState: State
let disabledState: State
```

Reply:
I mean the next
```swift 
extension SecureConversations.SendMessageButton {
    enum Props: Equatable {
        case normal(State)
        case disabled(State)
    }
}
extension SecureConversations.SendMessageButton.Props {
    struct State: Equatable {
        let title: String
        let titleFont: UIFont
        let foregroundColor: UIColor
        let backgroundColor: UIColor
        let borderColor: UIColor
        let borderWidth: Double
        let cornerRadius: Double
        let activityIndicatorColor: UIColor
        let isActivityIndicatorShown: Bool
    }
}
```

Reply:
No, the point is for them is to be exclusive at first place.

Reply:
To make it more clear: even though states are very similar, they are different type. That is by design, because at some point one of the sates may have different  property and generalisation will break. So I think it is ideal like that right now :)

Reply:
Ok, agree